Modules/arraymodule.c

```diff
@@ -221,7 +221,17 @@ b_setitem(arrayobject *ap, Py_ssize_t i, PyObject *v)
        the overflow checking */
     if (!PyArg_Parse(v, "h;array item must be integer", &x))
         return -1;
-    else if (x < -128) {
+
+    /* Check buffer validity after PyArg_Parse which may call user-defined
```

Comment:
Won't you always have this issue in *all* `*_setitem` formatters? none of them seem to check `ob_item`

Reply:
@picnixz Updated